base_layer/core/src/base_node/states/fetching_horizon_state.rs

```diff
@@ -111,6 +112,11 @@ impl HorizonInfo {
             return StateEvent::FatalError(format!("Synchronizing UTXO set failed. {}", e));
         }
 
+        info!(target: LOG_TARGET, "Validation synced horizon state.");
+        if let Err(e) = self.validate_horizon_state(shared).await {
+            return StateEvent::FatalError(format!("Validation of synced horizon state failed. {}", e));
```

Comment:
```suggestion
            return StateEvent::FatalError(format!("Validation of downloaded horizon state failed. {}", e));
```
We should probably try to recover from this rather than make it a fatal error, but we can make this a TODO

Reply:
Added a todo with a ticket number.